libsolidity/analysis/TypeChecker.cpp

```diff
@@ -805,12 +805,18 @@ void TypeChecker::endVisit(TryStatement const& _tryStatement)
 	TryCatchClause const& successClause = *_tryStatement.clauses().front();
 	if (successClause.parameters())
 	{
+		vector<TypePointer> exprTypes{_tryStatement.externalCall().annotation().type};
```

Comment:
Does it make sense to move this here? This should be a property ensured by the type checker and we check that this is indeed the case during code generation.